plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/_core.py

```diff
@@ -375,3 +281,49 @@ def normalize_key(key: str, str_to_replace: str, str_to_replace_with: str) -> st
     ]
     new_key = str_to_escape.join(updated_splits).replace("\\", "")
     return new_key
+
+
+def create_range_param_using_interval_override(override):
+    key = override.get_key_element()
+    value = override.value()
+    param = {
+        "name": key,
+        "type": "range",
+        "bounds": [value.start, value.end],
+    }
+    return param
+
+
+def create_choice_param_from_choice_override(override):
+    key = override.get_key_element()
+    value = override.value()
+    param = {
+        "name": key,
+        "type": "choice",
+        "values": value.list,
+    }
+    return param
+
+
+def create_choice_param_from_range_override(override):
+    key = override.get_key_element()
+    value = override.value()
+    param = {
+        "name": key,
+        "type": "choice",
+        "values": [val for val in value.range()],
```

Comment:
sweep_string_iterator().

If you need the values in the underlying type propose a change to the API to support it.
the string variant (sweep_string_iterator()) is designed to be used for generating coverrides.

Reply:
Maybe something like `override.sweep_value_iterator()` that returns an iterator over the values (while maintaining the correct value type).